docs/docs/debugging-gatsby.md

```diff
@@ -0,0 +1,132 @@
+
+```shell
+node --inspect-brk --no-lazy node_modules/gatsby/dist/bin/gatsby develop
+```
+
+ - `--inspect-brk` will enable the inspector agent which will allow you to connect a debugger. It will also pause execution until the debugger is connected and then wait for you to resume it.
+ - `--no-lazy` - this will force Node's V8 to disable lazy compilation and will help with using breakpoints
+
+### Connecting DevTools
+
+Open `chrome://inspect` in Chrome browser and connect to a "Remote Target" by clicking the `inspect` link:
+
+![Chrome inspect page](./images/chrome-devtools-inspect.png)
+
+You should see Chrome DevTools start and that code execution is paused at the start of the `gatsby.js` entry file:
+
+![Paused Chrome DevTools](./images/chrome-devtools-init.png)
+
+### Setting up `Sources`
+
+Right now you can't see your files in Sources. You need to add those using the "Add folder to workspace" button and pick the directory with the code you want to debug. If you want to debug code in your `gatsby-node.js` or your local plugins, pick your project directory. If you want debug the `gatsby` package you will have to pick the `gatsby` directory inside `node_modules`.
+
+This example has problematic code in your local `gatsby-node.js` file, so let's add the directory containing it to Sources. You should have a directory with your code in the left pane:
+
+![Files added to Sources tab](./images/chrome-devtools-files.png)
+
+### Using DevTools
+
+Let's go ahead and add a breakpoint just before the place that the error is thrown. To add a breakpoint navigate to `gatsby-node.js` and left click on a line number:
+
+![Added breakpoint](./images/chrome-devtools-new-breakpoint.png)
+
+Now you can resume code execution by clicking the "resume" icon in the DevTools debug toolbar (or press F8 on your keyboard). Gatsby will start running and pause once it reaches breakpoint allowing you to inspect variables:
+
+![Breakpoint hit](./images/chrome-devtools-breakpoint-hit.png)
+
+To inspect variables you can hover your mouse over them or go to the `Scope` section in the right-hand pane (either collapse the "Call Stack" section or scroll through it to the bottom).
+
+In our example `Node` is `undefined` and to figure out why, let's go backwards. `Node` is extracted from `args` so let's examine that by hovering `args`:
+
+![Examine variable](./images/chrome-devtools-examine-var.png)
+
+We can now see the problem - `args` doesn't contain `Node` - it contains `node`. So this small typographic mistake was causing our code to fail. Adjusting our code to use a lowercase `node` fixes the problem and we did that without adding tons of `console.log` output!
+
+### Finishing thoughts
+
+We can succussfully debug our code using Chrome DevTools but using it isn't really that convenient. There are a lot of steps we need to do manually everytime we want to use debugger, so in the next section you'll learn how to use the built-in debugging capabilities of VS Code.
```

Comment:
Maybe this could include a line about the good side of debugging with Chrome's dev tools? One is that you don't need to use a specific editor.

Reply:
That's good point, because right now it feels like we spent a lot of time showing DevTools debugger only to later mention that there are more conveniant options ;) I didn't know how to put it - one of main purposes and so detailed DevTools was that it should explain how node debugging work in general - so starting with `--inspect-brk` and connecting debugger - which is also what happens when you use debuggers in vscode or intelliJ suite - it's just hidden from you.